dandi/cli/cmd_upload.py

```diff
@@ -180,13 +192,15 @@ def upload(
 
     # Expand and validate all paths -- they should reside within dandiset
     orig_paths = paths
-    paths = list(find_dandi_files(paths))
+    paths = list(find_files(".*", paths) if allow_any_path else find_dandi_files(paths))
```

Comment:
should we allow `.*` (i.e. hidden files?) perhaps we restrict this to non-hidden files

Reply:
`.*` here is a regex, and hidden files by default are ignored by `find_files` - there is a dedicated option to allow for them.  I guess regex here could become `[^.].*` but currently effect would be the same, so imho ok to leave as is